src/gluonts/nursery/streaming_models/rnn/_estimator.py

```diff
@@ -0,0 +1,865 @@
+from ..native import ema
+from ..predictor import NETWORK_STATE_KEY
+from ..transform import (
+    AddShiftedTimestamp,
+    AnomalyScoringSplitter,
+    LeadtimeShifter,
+)
+from ._defaults import RnnDefaults
+from ._network import StreamingRnnPredictNetwork, StreamingRnnTrainNetwork
+
+LAGS_STATE_KEY = "s:lag"
+SCALE_STATE_KEY = "s:scale"
+FSC_STATE_KEY = "s:feat_static_cat"
+
+SUPPORTED_FREQS = ["1min", "5min", "10min", "1H", "1D"]
+
+DEFAULT_LAGS_UB = {
+    "1min": 1 * 7 * 24 * 60 + 3 * 60,  # one week + 3 hours,
+    "5min": 2 * 7 * 24 * 12 + 3 * 12,  # two weeks + 3 hours
+    "10min": 2 * 7 * 24 * 6 + 3 * 6,  # two weeks + 3 hours
+    "1H": 4 * 7 * 24 + 3,  # four weeks + 3 hours
+    "1D": 2 * 365,  # two year
+}
+
+DEFAULT_SMALL_LAGS = {
+    "1min": list(range(1, 61)),
+    "5min": list(range(1, 37)),
+    "10min": list(range(1, 25)),
+    "1H": list(range(1, 25)),
+    "1D": list(range(1, 22)),
+}
+
+
+class AddStreamScale(SimpleTransformation):
+    @validated()
+    def __init__(
+        self,
+        target_field: str,
+        scale_field: str,
+        state_field: str,
+        minimum_value: float,
+        initial_scale: float,
+        alpha: float,
+    ) -> None:
+        self.target_field = target_field
+        self.scale_field = scale_field
+        self.state_field = state_field
+        self.minimum_value = minimum_value
+        self.initial_scale = initial_scale
+        self.alpha = alpha
+
+    def transform(self, data: DataEntry) -> DataEntry:
+        target = data[self.target_field]
+        scale_state = data.get(self.state_field)
+
+        scale, new_scale_state = ema(
+            target,
+            alpha=self.alpha,
+            minimum_value=self.minimum_value,
+            initial_scale=self.initial_scale,
+            state=scale_state,
+        )
+
+        data[self.scale_field] = scale
+        data[self.state_field] = np.array(new_scale_state)
+        return data
+
+
+class AddLags(SimpleTransformation):
+    @validated()
+    def __init__(
+        self,
+        lag_seq: List[int],
+        lag_field: str,
+        target_field: str,
+        lag_state_field: str,
+    ) -> None:
+        self.lag_seq = sorted(lag_seq)
+        self.lag_field = lag_field
+        self.target_field = target_field
+        self.lag_state_field = lag_state_field
+        self.max_lag = self.lag_seq[-1]
+
+    def transform(self, data: DataEntry) -> DataEntry:
+        target = data[self.target_field]
+        buffer = data.get(self.lag_state_field)
+        if buffer is None:
+            t = np.concatenate([np.zeros(self.max_lag), target])
+        else:
+            t = np.concatenate([buffer, target])
+        lags = np.vstack(
+            [t[self.max_lag - l : len(t) - l] for l in self.lag_seq]
+        )
+        data[self.lag_field] = np.nan_to_num(lags)
+        data[self.lag_state_field] = t[-self.max_lag :]
+        return data
+
+
+class AddStreamAggregateLags(SimpleTransformation):
```

Comment:
Maybe we can move all the transformations to `stream_models/transform` so all the transformations are together? I guess there is no reason to split them here.